### PR TITLE
ev/change functional test framework

### PR DESCRIPTION
**Pull request by vilsbole**
- Remove Enzyme
- Setup react-testing-library

### CODE REVIEW

File: jest_config/jest.config.json [hunk 1]
```diff
@@ -2,25 +2,27 @@
...
   "moduleFileExtensions": ["ts", "tsx", "js", "jsx", "json", "worker.ts", "worker.tsx"],
   "moduleNameMapper": {
     "shared/(.*)": "<rootDir>/shared/$1",
+    "test-utils": "<rootDir>/jest_config/test-utils",
     "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/jest_config/__mocks__/fileMock.ts",
     "\\.(css|scss)$": "<rootDir>/jest_config/__mocks__/styleMock.ts",
     "\\.worker.(ts|tsx)": "<rootDir>/jest_config/__mocks__/workerMock.js"
   },
-  "testPathIgnorePatterns": ["node_modules", "dist"],
+  "testPathIgnorePatterns": [ "node_modules",  "dist" ],
```
Comment (Mrtenz) [inline jest_config/jest.config.json]: Can you remove the spaces here (to keep it consistent)?

File: jest_config/jest.config.json [hunk 2]
```diff
@@ -2,25 +2,27 @@
   "rootDir": "../",
   "verbose": true,
   "transform": {
-    "\\.tsx?$": "ts-jest"
+    "^.+\\.jsx?$": "babel-jest",
+    "^.+\\.tsx?$": "ts-jest"
```
Comment (Mrtenz) [inline jest_config/jest.config.json]: We can remove `ts-jest` and just use `babel-jest`, e.g.
```
"^.+\\.[t|j]sx?$": "babel-jest"
```
Babel is configured for TS support already.

File: jest_config/test-utils.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options })
+
+
+// re-export everything
+export * from '@testing-library/react'
+
+// override render method
+// export { customRender as render }
```
Comment (Mrtenz) [inline jest_config/test-utils.js]: Is this planned for another PR? Currently `customRender` is not exported.

Comment (vilsbole) [reply jest_config/test-utils.js]: Yes it's in anticipation for when we start testing components and need to wrap all the providers.

File: jest_config/jest.config.json (hunk 1 above)
Comment (vilsbole) [reply jest_config/jest.config.json]: Done

File: jest_config/jest.config.json (hunk 2 above)
Comment (vilsbole) [reply jest_config/jest.config.json]: Following your suggestion it has been removed in favor of defaults

File: tslint.json [hunk 4]
```diff
@@ -33,4 +33,4 @@
     "react-no-dangerous-html": true
   },
   "rulesDirectory": ["node_modules/tslint-microsoft-contrib"]
-}
+}
```
Comment (Mrtenz) [inline tslint.json]: 🤔

File: jest_config/jest.config.json [hunk 5]
```diff
@@ -1,31 +1,24 @@
...
   "moduleFileExtensions": ["ts", "tsx", "js", "jsx", "json", "worker.ts", "worker.tsx"],
   "moduleNameMapper": {
     "shared/(.*)": "<rootDir>/shared/$1",
+    "test-utils": "<rootDir>/jest_config/test-utils",
     "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/jest_config/__mocks__/fileMock.ts",
     "\\.(css|scss)$": "<rootDir>/jest_config/__mocks__/styleMock.ts",
     "\\.worker.(ts|tsx)": "<rootDir>/jest_config/__mocks__/workerMock.js"
   },
-  "testPathIgnorePatterns": ["node_modules", "dist"],
+  "testPathIgnorePatterns": ["node_modules",  "dist"],
```
Comment (Mrtenz) [inline jest_config/jest.config.json]: Unnecessary whitespace.

File: jest_config/jest.config.json [hunk 6]
```diff
@@ -1,31 +1,24 @@
 {
   "rootDir": "../",
   "verbose": true,
-  "transform": {
-    "\\.tsx?$": "ts-jest"
-  },
   "testRegex": "(/__tests__/.*|(\\.|/)(test|spec))\\.(jsx?|tsx?)$",
-  "moduleDirectories": ["node_modules", "common"],
+  "moduleDirectories": ["node_modules", "common"],
```
Comment (Mrtenz) [inline jest_config/jest.config.json]: Unnecessary whitespace.

File: tslint.json (hunk 4 above)
Comment (vilsbole) [reply tslint.json]: Absolutely no clue :(

File: jest_config/jest.config.json (hunk 5 above)
Comment (vilsbole) [reply jest_config/jest.config.json]: Done

File: jest_config/jest.config.json (hunk 6 above)
Comment (vilsbole) [reply jest_config/jest.config.json]: Done
